### PR TITLE
Fix bug if no value is supplied.

### PR DESCRIPTION
**Pull request by martindale**
This fixes a minor bug in some corner cases where multiple network changes may take place over time, or specifically, if the network is undefined.

### CODE REVIEW

File: test/networks.js [hunk 1]
```diff
@@ -40,6 +40,25 @@ describe('Networks', function() {
...
+      xprivkey: 0x0278ade5,
+      networkMagic: 0xe7beb4d5,
+      port: 20008,
+      dnsSeeds: [
+        'somenet.localhost'
+      ]
+    };
+    networks.add(custom);
+    var network = networks.get(undefined);
+    should.not.exist( network );
```
Comment (braydonf) [inline test/networks.js]: whitespace `should.not.exist(network)`

File: test/networks.js [hunk 2]
```diff
@@ -40,6 +40,25 @@ describe('Networks', function() {
...
     }
   });
+  
+  it('should not set a network map for an undefined value', function() {
+    var custom = {
+      name: 'somenet',
+      pubkeyhash: 0x13,
+      privatekey: 0x93,
+      scripthash: 0x11,
+      xpubkey:  0x0278b20f,
```
Comment (braydonf) [inline test/networks.js]: extra space here

File: lib/networks.js [hunk 3]
```diff
@@ -75,7 +75,7 @@ function addNetwork(data) {
   });
 
   _.each(_.values(network), function(value) {
-    if (!_.isObject(value)) {
+    if (value && !_.isObject(value)) {
```
Comment (braydonf) [inline lib/networks.js]: `if (!_.isUndefined(value) && !_.isObject(value)) {`

to handle the case between `0` and `undefined`

**Comment by maraoz** [comment]
@martindale tests seem to be broken. Please check. I've restarted travis just in case it's gone mad

**Comment by braydonf** [comment]
ACK

File: test/networks.js [hunk 4]
```diff
@@ -40,6 +40,25 @@ describe('Networks', function() {
...
+      xprivkey: 0x0278ade5,
+      networkMagic: 0xe7beb4d5,
+      port: 20008,
+      dnsSeeds: [
+        'somenet.localhost'
+      ]
+    };
+    networks.add(custom);
+    var network = networks.get(undefined);
+    should.not.exist(network);
```
Comment (maraoz) [inline test/networks.js]: shouldn't `networks.get(undefined)` throw an exception?

Comment (braydonf) [inline test/networks.js]: I don't think so currently, since there is code that is written like this:
`var network = Networks.get(argument) || Networks.defaultNetwork;` 
If argument is `undefined`, `undefined` is returned and then is set to the default.

Comment (maraoz) [inline test/networks.js]: I see. I'll merge then
